backend/tarsy/models/mcp_selection_models.py

```
@@ -29,20 +29,82 @@ class MCPServerSelection(BaseModel):
-        min_length=1,
-        description="List of selected MCP servers with optional tool filtering"
+        description="List of selected MCP servers with optional tool filtering. Empty list allowed for tool-less agents."
+    )
+    native_tools: Optional[NativeToolsConfig] = Field(
+        None,
+        description="Optional native tools override for Google/Gemini models. "
+                    "When specified, completely replaces provider default settings."
     )
+    
+    @model_serializer(mode='wrap', when_used='always')
+    def _serialize_model(self, serializer: Any, info: Any) -> Dict[str, Any]:
```

Ruff (0.14.5)

Unused method argument: `info`

(ARG002)